samcli/hook_packages/terraform/copy_terraform_built_artifacts.py

```diff
@@ -268,3 +264,61 @@ def cli_exit():
     except OSError as ex:
         LOG.error("Copy/Unzip unsuccessful!", exc_info=ex)
         cli_exit()
+
+
+def create_backend_override():
+    """
+    Copies and rename the override tf file from the metadata directory to the root
+    directory of the TF application.
+    """
+    override_src_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), TF_BACKEND_OVERRIDE_FILENAME)
+    override_dest_path = os.path.join(os.getcwd(), TF_BACKEND_OVERRIDE_FILENAME + ".tf")
+    shutil.copyfile(override_src_path, override_dest_path)
+
+
+if __name__ == "__main__":
+    # Gather inputs and clean them
+    argparser = argparse.ArgumentParser(
+        description="Copy built artifacts referenced in a json file " "(passed via stdin) matching a search pattern"
+    )
+    argparser.add_argument(
+        "--expression",
+        type=str,
+        required=True,
+        help="Jpath query expression separated by | (delimiter) "
+        "and allows for searching within a json object."
+        "eg: |values|sub_module|[?address==me]|output",
+    )
+    argparser.add_argument(
+        "--directory",
+        type=str,
+        required=True,
+        help="Directory to which extracted expression " "contents are copied/unzipped to",
+    )
+    argparser.add_argument(
+        "--target",
+        type=str,
+        required=True,
+        help="Terraform resource path for the SAM CLI Metadata resource",
+    )
+
+    arguments = argparser.parse_args()
+    directory_path = os.path.abspath(arguments.directory)
+    expression = arguments.expression
+    target = arguments.target
+
+    create_backend_override()
+
+    subprocess.check_call(["terraform", "init", "-reconfigure"])
```

Comment:
kind of obvious q: do we already know if we check for the terrraform binary to exist and throw an error message before we ever reach this stage?

Reply:
This script is used only when running `sam build --hook-package-id terraform`. Before this script is invoked (through LB CustomMakeWorkflow), the Terraform Hook has already run `terraform init` and `terraform plan` during the prepare phase. So if the terraform binary doesn't exist for any reason, the error should have been thrown earlier.